logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SyncManager.kt

```diff
@@ -1,43 +1,176 @@
 package com.wire.kalium.logic.sync
 
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.data.event.Event
+import com.wire.kalium.logic.data.event.EventRepository
 import com.wire.kalium.logic.data.sync.SyncRepository
 import com.wire.kalium.logic.data.sync.SyncState
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.logic.functional.onFailure
+import com.wire.kalium.logic.functional.onSuccess
+import com.wire.kalium.logic.kaliumLogger
+import com.wire.kalium.util.KaliumDispatcher
+import io.ktor.utils.io.errors.IOException
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
 import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
 
 interface SyncManager {
-    fun completeSlowSync()
-    suspend fun waitForSlowSyncToComplete()
+    fun onSlowSyncComplete()
+
+    /**
+     * Blocks the caller until all pending events are processed.
+     */
+    suspend fun waitForSyncToComplete()
     suspend fun isSlowSyncOngoing(): Boolean
     suspend fun isSlowSyncCompleted(): Boolean
+    fun onSlowSyncFailure(cause: CoreFailure): SyncState
 }
 
 class SyncManagerImpl(
     private val workScheduler: WorkScheduler,
-    private val syncRepository: SyncRepository
+    private val eventRepository: EventRepository,
+    kaliumDispatcher: KaliumDispatcher,
+    private val syncRepository: SyncRepository,
+    private val conversationEventReceiver: ConversationEventReceiver
 ) : SyncManager {
```

Comment:
```suggestion
    private val eventRepository: EventRepository,
    private val syncRepository: SyncRepository,
    private val conversationEventReceiver: ConversationEventReceiver
        kaliumDispatcher: KaliumDispatcher = KaliumDispatcherImpl,
) : SyncManager {
```
KaliumDispatcher can be injected into constructor